### PR TITLE
[Merged by Bors] - feat(RingTheory/DedekindDomain/IntegralClosure): Added `IsIntegralClosure.finite`

### PR DESCRIPTION
**Pull request by 4hma4d**
Let A be a Noetherian integrally closed domain with fraction field K. Let L/K be a finite separable field extension. Then the integral closure of A in L is finite over A.
The proof immediately follows from `IsIntegralClosure.isNoetherian`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/DedekindDomain/IntegralClosure.lean [hunk 1]
```diff
@@ -185,6 +185,14 @@ theorem IsIntegralClosure.isNoetherianRing [IsIntegrallyClosed A] [IsNoetherianR
     IsNoetherianRing C :=
   isNoetherianRing_iff.mpr <| isNoetherian_of_tower A (IsIntegralClosure.isNoetherian A K L C)
 
+/-- If `L` is a finite separable extension of `K = Frac(A)`, where `A` is
+integrally closed and Noetherian, the integral closure `C` of `A` in `L` is
+finite over `A`. -/
+theorem IsIntegralClosure.isFinite [IsIntegrallyClosed A] [IsNoetherianRing A] :
```
Comment (Ruben-VandeVelde) [inline Mathlib/RingTheory/DedekindDomain/IntegralClosure.lean:191]:
```suggestion
theorem IsIntegralClosure.finite [IsIntegrallyClosed A] [IsNoetherianRing A] :
```
and fix the PR title. The conclusion is `Finite`, not `IsFinite`

**Review by Ruben-VandeVelde — COMMENTED** [review]
Thanks!

**Comment by kbuzzard** [comment]
Rather than fiddling with the labels, I should just ask: is this ready to go now? I'm happy to merge this but it's still labelled awaiting-author.

**Comment by erdOne** [comment]
Do we really need this? Isn't this strictly weaker than `IsIntegralClosure.isNoetherian`?

**Comment by kbuzzard** [comment]
Sure it is. But I wanted it in FLT, and `exact?` didn't find it, and I didn't know what to do next.

**Comment by 4hma4d** [comment]
So is there a verdict on if this should be added?

**Review by jcommelin — COMMENTED** [review]
If the `haveI` is needed, then we should add this to mathlib.

Thanks!

bors merge

**Comment by kbuzzard** [comment]
> If the `haveI` is needed, then we should add this to mathlib.

It is needed, because typeclass inference can't find `IsNoetherian A C` even though `IsNoetherian` is a class, because `IsIntegralClosure.isNoetherian` isn't allowed to be an instance (as it uses the fields of fractions `K` and `L` so it's a dangerous instance)

**Pull request closed without merging**